hydra-node/src/Hydra/Ledger/Cardano.hs

```diff
@@ -144,8 +144,8 @@ mkSimpleTx ::
   SigningKey PaymentKey ->
   Either TxBodyError Tx
 mkSimpleTx (txin, TxOut owner valueIn datum refScript) (recipient, valueOut) sk = do
-  body <- createAndValidateTransactionBody bodyContent
-  let witnesses = [makeShelleyKeyWitness body (WitnessPaymentKey sk)]
+  body <- createAndValidateTransactionBody BabbageEra bodyContent
+  let witnesses = [makeShelleyKeyWitness ShelleyBasedEraBabbage body (WitnessPaymentKey sk)]
```

Comment:
Commenting once, but applies to multiple places:

Avoid mentioning `Babbage` in packages outside of `hydra-cardano-api`. Currently, that package serves as an "anti-corruption" layer and hard-codes the era we target. Even if we never make it configurable, we want to limit the amount of code that needs changing. That's also the reason why we have all those patterns and alias functions in `Hydra.Cardano.Api`, which remove the need for call sites to pick an era.

Must: try to avoid "witness" values outside of `hydra-cardano-api`.

Concretely telling from this call site, I would propose to provide a fixed-era `createAndValidateTransactionBody :: TxBodyContent Era -> Either TxBodyError (TxBody Era)` in `hydra-cardano-api` and the same for `makeShelleyKeyWitness` and/or using type class constraints. Note the `Tx` return value of `mkSimpleTx` is a hard-coded `Tx Era = Tx BabbageEra` right now.